Replace relative imports with absolute ones, run Black on code base

dcambie opened the pull request:
Somehow, relative imports were causing issues so now they are all absolute to the position within the octopus package.

Also, black was run on the codebase (which is the reason why the changeset looks so large. I can split that commit to a different PR if needed.

An example of HW interaction with an R2R4 is also added (it works, yay!) as a minimum working example

dcambie committed:
Run black on repo

dcambie committed:
Run black on repo

dcambie committed:
Make it pip-installable

Fix path for blocks (now in blocktopus)

dcambie committed:
Make it pip-installable

Fix path for blocks (now in blocktopus)

dcambie committed:
BUGFIX: Python2 to python3 change

dcambie committed:
BUGFIX: Python2 to python3 change

dcambie committed:
BUGFIX: Python2 to python3 change

dcambie committed:
BUGFIX: Python2 to python3 change

dcambie committed:
Make relative import absolute (fixes circular import loop)

dcambie committed:
Make relative import absolute (fixes circular import loop)

dcambie committed:
Add Vapourtec stuff to gitignore

To be sure not to commit them by mistake

dcambie committed:
Add Vapourtec stuff to gitignore

To be sure not to commit them by mistake

dcambie committed:
Final changes to imports

dcambie committed:
Final changes to imports

dcambie committed:
README updates

dcambie committed:
README updates

dcambie committed:
Include missing Exception class

dcambie committed:
Include missing Exception class

dcambie committed:
Add working (yay!) HW control example

dcambie committed:
Add working (yay!) HW control example

richardingham committed:
Create the octopus.blocks namespace package.

richardingham commented on requirements.txt:
```diff
@@ -9,3 +9,5 @@ pyasn1
 wget
 opencv-python
+Phidgets
+SimpleCV
```
Does the Phidgets lib definitely work with Python3?  I'm pretty sure SimpleCV is Python 2.7 only and no longer maintained. I am in the process of removing the SimpleCV dependency (see octopus/image/functions.py).

dcambie replied:
> Yes, could you please split out the re-formatting into a different PR?

Formatting changes are unfortunately at the base of this branch, so I've moved to a different PR and I'll rebase on master to clean this once those are merged. Sorry for that.

> Does the Phidgets lib definitely work with Python3? I'm pretty sure SimpleCV is Python 2.7 only and no longer maintained. I am in the process of removing the SimpleCV dependency (see octopus/image/functions.py).

The latest Phidgets API release (Phidget22) seems to support Python3.
[Phidgets22 @ Pypi]([](https://pypi.org/project/Phidget22/))

I have not tested it yet, but since we have some pressure sensors with 4..20mA connection and no reader yet I have just ordered a Phidget hub+4..20mA adapter to test it and I'll report on that in the future.

richardingham replied:
Phidgets - cool, thanks, I look forward to the result!
SimpleCV - the project is dead so please don't add SimpleCV here. The current opencv implementation is badly documented but functional!

richardingham replied:
> Formatting changes are unfortunately at the base of this branch, so I've moved to a different PR and I'll rebase on master to clean this once those are merged. Sorry for that.

You should be able to do a rebase to separate the two branches - [see the howto here](https://gist.github.com/unbracketed/889c844473bcca1917e2).

richardingham commented on setup.py:
```diff
@@ -14,7 +14,7 @@
     long_description_content_type="text/markdown",
     url="https://github.com/richardingham/octopus",
-    packages=["octopus", "octopus.manufacturer", "octopus.blocks"],
+    packages=["octopus", "octopus.manufacturer", "octopus.blocktopus.blocks"],
```
I think this is fixed by 53a9188 but if you could test on your system that would be great. 
octopus.blocks is a new namespace package for plugins. octopus.blocktopus.blocks is the existing blocks.

dcambie replied:
Indeed,fixed by https://github.com/richardingham/octopus/commit/53a91880606b3551ef3c7ac7b944c07c84807f83, rolled back now

richardingham commented on .gitignore:
```diff
@@ -67,3 +67,5 @@ twistd.pid
 
 .DS_Store
+/octopus/manufacturer/vapourtec.py
+/octopus/blocktopus/blocks/vapourtec.py
```
Can you keep this in your own repository please and not commit it? 
This is meant to be the purpose of plugins, though that system should probably be improved :-) (or documented...)

dcambie replied:
> Can you keep this in your own repository please and not commit it?
> This is meant to be the purpose of plugins, though that system should probably be improved :-) (or documented...)

Sure, actually I use those as plugins but sometimes is faster to just copy and paste and I added those to gitignore as to be 100% sure that they never get committed by mistake as once they are in the git history it is hard to remove them

richardingham replied:
Ah - that makes sense! Still, this shouldn't go into the main repository. I'd recommend putting a .gitignore file in a subdirectory (either in octopus/, or in both octopus/manufacturer and octopus/blocks) with these in and then not committing that .gitignore.

richardingham commented on octopus/protocol/basic.py:
```diff
@@ -10,7 +10,7 @@
 
 # Package Imports
-from octopus.queue import AsyncQueue, AsyncQueueRetry
+from octopus.octopus_queue import AsyncQueue, AsyncQueueRetry
```
I'm not sure I understand this - the module is octopus.queue.

dcambie replied:
Fixed in https://github.com/richardingham/octopus/pull/9/commits/5bd3b5ab4e2ea9c7ca56064bc239136136fd72ea

I had renamed `queue.py` into `octopus_queue.py`, iirc because of some namespace collision. However that does not seem to be an issue anymore so back to old naming now

richardingham reviewed (changes requested):
Thanks for all your work here - it's great to see that this part is working again! 

Yes, could you please split out the re-formatting into a different PR?

Commit:
Revert "Make it pip-installable"

This reverts commit b6830dcd

Commit:
Merge remote-tracking branch 'origin/black_etc' into black_etc

# Conflicts:
#	setup.py

Commit:
Rename back octopus_queue.py to queue.py

I think I had changed this for some namespace collision issue,
but that does not seem to be a problem anymore so...
